apps/cowswap-frontend/src/modules/volumeFee/state/volumeFeeAtom.ts

```diff
@@ -1,26 +1,64 @@
 import { atom } from 'jotai'
 
+import { getCurrencyAddress } from '@cowprotocol/common-utils'
 import { walletInfoAtom } from '@cowprotocol/wallet'
 import { resolveFlexibleConfig, TradeType as WidgetTradeType } from '@cowprotocol/widget-lib'
 
 import { injectedWidgetPartnerFeeAtom } from 'modules/injectedWidget'
-import { tradeTypeAtom } from 'modules/trade'
+import { derivedTradeStateAtom, tradeTypeAtom } from 'modules/trade'
 import { TradeType } from 'modules/trade/types/TradeType'
 
 import { cowSwapFeeAtom } from './cowswapFeeAtom'
 import { safeAppFeeAtom } from './safeAppFeeAtom'
+import { taxFreeAssetsAtom } from './taxFreeAssetsAtom'
 
 import { VolumeFee } from '../types'
 
 export const volumeFeeAtom = atom<VolumeFee | undefined>((get) => {
+  const tradeState = get(derivedTradeStateAtom)
   const cowSwapFee = get(cowSwapFeeAtom)
   const widgetPartnerFee = get(widgetPartnerFeeAtom)
   const safeAppFee = get(safeAppFeeAtom)
+  const shouldSkipFee = get(shouldSkipFeeAtom)
+
+  if (!widgetPartnerFee && shouldSkipFee) {
```

Comment:
So this applies before the cowSwapFee, where we use the hardcoded list of tokens that should skip the fee.
Meaning, when `shouldSkipFee` is false, the fee can still be skipped.

Not sure what's the best approach.
That can work as a fallback, but can also easily be out of sync.

Reply:
I would go this way:
 - release this feature and the hardcoded list of stablecoins
 - add the stablecoins into CMS
 - remove the hardcode